poof.py

```diff
@@ -119,30 +122,54 @@ def package_remove(package_id, force=True, verbose=False):
             os.remove(path)
         except OSError as e:
             clean = False
-            print e
+            print(e)
     dirs = [prefix + x for x in dirs]
-    dirs.sort(lambda p1, p2: p1.count('/') - p2.count('/'),
```

Comment:
you can write your own sorting method using basic concepts of the language to make it compatible between versions

```python
   # EXAMPLE
   
   dirs = ['file/example/dddd/222', 'file/example/dddd/222/55/ffd', 'file/dd', 'file/example/aaa']
   ordered_dirs = [
        dir_set[1] for dir_set in 
        sorted({directory.count('/'):directory for directory in original}.items())
   ]
   ordered_dirs.reverse()

   # ordered_dirs => ['file/example/dddd/222/55/ffd', 'file/example/dddd/222', 'file/example/aaa', 'file/dd']
  
```

Reply:
Can you test your idea with this input? It seems to me that it will chop paths `'file/example/aaa'` and `'file/example/aba'` into just one thing.

```python
dirs = [
    'file/example/dddd/222',
    'file/example/dddd/222/55/ffd',
    'file/aa',
    'file/example/aaa',
    'file/example/aba',
    'file/dd'
]
```

This is what I've been playing…

```Python
#!/usr/local/bin/python3

dirs = [
    'file/example/dddd/222',
    'file/example/dddd/222/55/ffd',
    'file/aa',
    'file/example/aaa',
    'file/example/aba',
    'file/dd'
]

#new_dirs = [y[1] for y in sorted({x.count('/'):x for x in dirs}.items(), reverse=True)]
print(len(dirs))
print(dirs)
#new_dirs = sorted({x.count('/'):x for x in dirs}.items(), reverse=True)
#dirs = sorted(dirs, key=lambda p: p.count('/'), reverse=True)

dirs = {(x.count('/'), x) for x in dirs}
print(dirs)
dirs = sorted(dirs, key=lambda x: x[0], reverse=True)
print(len(dirs))
print(dirs)
```